### DISCUSSION TITLE
Allow soft deletable models to be implicitly bound

### DISCUSSION THREAD

**Discussion by lpointet** [livewire/livewire#5395] [3. Feature ideas]
Hello,

I'm trying to have a Livewire component actions to implicitly bind my model. This is working perfectly as soon as they are regular models, but I have one model which is soft deletable. The thing is that I'm allowing some of my users to see already (soft-)deleted objects, and perform actions on them.

Unfortunately, in this case, the implicit binding doesn't work and it seems there is no simple way to configure it in my component.

With the Laravel Resource Controller routing system, I'm using the following syntax:
`Route::resource('name', Model::class)->withTrashed();`

I found out the following code in Illuminate/Routing/ImplicitRouteBinding.php:
```
$routeBindingMethod = $route->allowsTrashedBindings() && in_array(SoftDeletes::class, class_uses_recursive($instance))
                        ? 'resolveSoftDeletableRouteBinding'
                        : 'resolveRouteBinding';
```

So could Livewire make use of a similar feature in the implicit binding system?

I can see in the file ImplicitlyBoundMethod.php that only `resolveRouteBinding` is used to retrieve the model, but not `resolveSoftDeletableRouteBinding`.

For now, as a workaround, I'm simply overriding the `resolveRouteBindingQuery` method of my model to leverage the use of trashed ones.

Thanks!

**Comment by joshhanley** [comment]
@lpointet yeah makes sense to me. Normally I would suggest submitting a PR, but currently we're trying to limit changes to V2 while we work on V3 to allow the transition to be a smooth as possible. So I would suggest that anyone who is willing to submit a PR for this feature, to wait until V3 has been released 🙂

**Comment by marcheffels** [comment]
I have the same problem. Due to time constraints, I can't wait for version 3. 
I hope it is not rated as a feature but as a bug.

@joshhanley  I'm willing to submit a little PR. Is there a chance my PR will be merged?
